src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -132,7 +132,7 @@ static ss::future<list_offset_partition_response> list_offsets_partition(
     auto res = co_await kafka_partition->timequery(storage::timequery_config{
       kafka_partition->start_offset(),
       timestamp,
-      offset,
+      model::prev_offset(offset),
```

Comment:
> kafka: fix off-by-one in timequery
> Not easy to test that this is right so not going to for now.

@nvartolomei please include in commit message more detail. the commit by inspection implies an off-by-one error, but there are critically useful pieces of information a reader in the future may want. for example, how was it discovered, what are symptoms of it being wrong, etc...